allennlp/tests/training/trainer_test.py

```diff
@@ -120,6 +120,21 @@ def test_passing_trainer_multiple_gpus_raises_error(self):
             Trainer(
                 self.model, self.optimizer, self.data_loader, num_epochs=2, cuda_device=[0, 1],
             )
+    
+    @pytest.mark.skipif(not torch.cuda.is_available(), reason="No CUDA device registered.")
+    @pytest.mark.skipif(importlib.util.find_spec("apex") is None, reason="Apex is not installed.")
+    def test_trainer_can_run_amp(self):
+        from apex import amp
+        self.model.cuda()
+        trainer = Trainer(
+            self.model, self.optimizer, self.data_loader, num_epochs=2, cuda_device=0, opt_level="O1"
+        )
+        metrics = trainer.train()
+        assert "peak_cpu_memory_MB" in metrics
+        assert isinstance(metrics["peak_cpu_memory_MB"], float)
+        assert metrics["peak_cpu_memory_MB"] > 0
+        assert "peak_gpu_0_memory_MB" in metrics
+        assert isinstance(metrics["peak_gpu_0_memory_MB"], int)
```

Comment:
Added a test to make sure the model runs with `amp`. There are two `skipif` checks, one to make sure a cuda device is avaible and one to check if `apex` is importable.